types/tailwindcss/tailwind-config.d.ts

```diff
@@ -0,0 +1,1188 @@
+export type Variant =
+    | 'responsive'
+    | 'first'
+    | 'last'
+    | 'odd'
+    | 'even'
+    | 'visited'
+    | 'checked'
+    | 'group-hover'
+    | 'group-focus'
+    | 'focus-within'
+    | 'hover'
+    | 'focus'
+    | 'focus-visible'
+    | 'active'
+    | 'disabled'
+    | 'dark';
+
+export interface TailwindColorGroup {
```

Comment:
How does this extend to config that has been extended/overwritten with a custom solution? I'm assuming I can't do e.g. `tailwindConfig.theme.colors.main.dark` even if that's how my tailwind.config.js is set up?

Reply:
It can, to a certain extent

I have added a few things which will allow for extention where appropriate:

```ts
readonly [key: string]: string | TailwindColorGroup;
```

Reply:
This change doesn't seem to have been made...

Reply:
done, added,  `TailwindColorGroup` now has  `readonly [key: string]: string;`

please check

Reply:
Thanks!  I believe this addresses @themagickoala's concerns.

Reply:
There should ideally be a dynamic key in TailwindColorConfig which can be a string or TailwindColorGroup.